tonic/models/vic/grid_params.py

```diff
@@ -661,64 +1218,172 @@ def grid_params(soil_dict, target_grid, snow_dict, veg_dict, veglib_dict,
                 for y, x in pyzip(ymask, xmask):
                     out_dict[var][:, :, y, x] = fill_val
 
+            elif mydict[var].ndim == 4:
+                j = mydict[var].shape[1]
+                k = mydict[var].shape[2]
+                l = mydict[var].shape[3]
+                out_dict[var] = np.ma.zeros((j, k, l, ysize, xsize),
+                                            dtype=dtype)
```

Comment:
Thanks.  Note: when I say "someone else's inefficient code" I don't in any way mean to imply that I'm a great python coder.  I'm just saying that I hadn't budgeted for doing an overhaul of the code, and hadn't known at the time (last year) that there was anything in the existing code that needed overhauling.